zstash/utils.py

```diff
@@ -5,10 +5,69 @@
 import sqlite3
 import subprocess
 from datetime import datetime, timezone
+from enum import Enum
 from fnmatch import fnmatch
 from typing import Any, List, Tuple
-
-from .settings import TupleTarsRow, config, logger
+from urllib.parse import urlparse
+
+from .settings import DEFAULT_CACHE, TupleTarsRow, config, logger
+
+
+class HPSSType(Enum):
+    NO_HPSS = 1
+    SAME_MACHINE_HPSS = 2
+    GLOBUS = 3
+
+
+class CommandInfo(object):
+    def __init__(self, command_name: str):
+        self.command_name = command_name
+
+        # Directories
+        self.dir_called_from = os.getcwd()
+        self.dir_to_archive = None
+        self.cache_dir = DEFAULT_CACHE
+
+        # HPSS
+        # self.hpss_type # Use set_hpss_parameters
+        # self.hpss_path # Use set_hpss_parameters
+
+        # Globus-specific
+        # self.globus_path # Use set_hpss_parameters
+        # remote_endpoint = None
+        # local_endpoint = None
+        # transfer_client = None
+        # transfer_data = None
+        # task_id = None
+        # archive_directory_listing = None
```

Comment:
Perhaps this could itself be pulled out into a `GlobusInfo` class that is used to pass around Globus specific state.